### PR TITLE
Update README.md 

### PR DESCRIPTION
**Pull request by ZiTao-Li**
## Description

* Update README.md to make it more clear and easy to follow.
* Update search.py to fix the typo in doc-string
* Update description for ollama, gemini and dashscope API
* Update model related documents
* Re-organize & rename example directories

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: docs/sphinx_doc/zh_CN/source/tutorial_zh/203-model.md [hunk 1]
```diff
@@ -87,69 +90,301 @@ class OpenAIChatWrapper(OpenAIWrapper):
...
+    "client_args": {                            # 用于初始化OpenAI API Client的参数
+        # e.g. "max_retries": 3,
     },
-    "generate_args": {              # 调用模型时传入的参数
-        "temperature": 0.0
+    "generate_args": {                          # 模型API接口被调用时传入的参数
+        # e.g. "temperature": 0.0
     },
-    "budget": 100.0                 # API费用预算
+    "budget": 100                               # API budget
```
Comment (pan-x-c) [inline docs/sphinx_doc/zh_CN/source/tutorial_zh/203-model.md:136]: API budget -> API 费用预算（budget）

File: docs/sphinx_doc/zh_CN/source/tutorial_zh/203-model.md [hunk 2]
```diff
@@ -87,69 +90,301 @@ class OpenAIChatWrapper(OpenAIWrapper):
...
 
     # 可选参数
-    "api_key": "xxx",               # OpenAI API Key，如果没有提供则会从环境变量中读取
-    "organization": "xxx",          # 组织名称，如果没有提供则会从环境变量中读取
-    "client_args": {                # 初始化OpenAI API Client的参数
-        "max_retries": 3,
+    "api_key": "{your_api_key}",                # OpenAI API Key，如果没有提供，将从环境变量中读取
+    "organization": "{your_organization}",      # Organization name，如果没有提供，将从环境变量中读取
+    "client_args": {                            # 用于初始化OpenAI API Client的参数
+        # e.g. "max_retries": 3,
```
Comment (pan-x-c) [inline docs/sphinx_doc/zh_CN/source/tutorial_zh/203-model.md:131]: e.g. -> 例如

建议不要中英文混杂

File: docs/sphinx_doc/zh_CN/source/tutorial_zh/203-model.md [hunk 3]
```diff
@@ -87,69 +90,301 @@ class OpenAIChatWrapper(OpenAIWrapper):
...
+
+```python
+{
+    "config_name": "my_postapiwrapper_config",
     "model_type": "post_api",
 
-    # 详细参数
-    "api_url": "http://xxx.png",
+    # 必要参数
+    "api_url": "https://xxx.com",
```
Comment (pan-x-c) [inline docs/sphinx_doc/zh_CN/source/tutorial_zh/203-model.md:373]: -> https://xxx.xxx

不一定是 com 结尾，不要引起误导

File: docs/sphinx_doc/en/source/tutorial/203-model.md [hunk 4]
```diff
@@ -93,68 +70,298 @@ class OpenAIChatWrapper(OpenAIWrapper):
...
+
+```python
+{
+    "config_name": "my_postapiwrapper_config",
     "model_type": "post_api",
 
-    # Detailed parameters
-    "api_url": "http://xxx.png",
+    # Required parameters
+    "api_url": "https://xxx.com",
```
Comment (pan-x-c) [inline docs/sphinx_doc/en/source/tutorial/203-model.md:353]: -> https://xxx.xxx

File: README.md [hunk 5]
```diff
@@ -12,66 +14,116 @@ English | [**中文**](README_ZH.md)
...
+
+---
+
+## What's AgentScope?
+
+AgentScope is an innovative multi-agent platform designed to empower developers
+to build multi-agent applications with large-scale models.
+It features three high-level capabilities:
+
+- 🤝 **Easy-to-Use**: Designed for developers, with [fruitful components](),
```
Comment (pan-x-c) [inline README.md:49]: add links for fruitful components

File: README.md [hunk 6]
```diff
@@ -12,66 +14,116 @@ English | [**中文**](README_ZH.md)
...
+|----------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|
 | <img src="https://gw.alicdn.com/imgextra/i1/O1CN01hhD1mu1Dd3BWVUvxN_!!6000000000238-2-tps-400-400.png" width="100" height="100"> | <img src="https://img.alicdn.com/imgextra/i2/O1CN01tuJ5971OmAqNg9cOw_!!6000000001747-0-tps-444-460.jpg" width="100" height="100"> | <img src="https://img.alicdn.com/imgextra/i3/O1CN01UyfWfx1CYBM3WqlBy_!!6000000000092-2-tps-400-400.png" width="100" height="100"> |
 
 ----
 
 ## News
+- ![new](https://img.alicdn.com/imgextra/i4/O1CN01kUiDtl1HVxN6G56vN_!!6000000000764-2-tps-43-19.png)
+[2024-03-15] We release **AgentScope** v0.0.2 now! In this new version,
+AgentScope supports [DashScope](), [ollama]() and [Gemini]() APIs.
```
Comment (pan-x-c) [inline README.md:31]: add links for dashscope, olllama and gemini

File: README.md [hunk 7]
```diff
@@ -12,66 +14,116 @@ English | [**中文**](README_ZH.md)
...
+|                        | Text Embedding  | `DashScopeTextEmbeddingWrapper`  |
+| Post Request based API | -               | `PostAPIModelWrapper`            |
+
+
+**Supported Local Model Deployment**
+
+AgentScope enables developers to rapidly deploy local model services using
+the following libraries.
+
+- [ollama (CPU inference)]()
```
Comment (pan-x-c) [inline README.md:84]: add links here

File: scripts/README.md [hunk 8]
```diff
@@ -1,36 +1,108 @@
...
+    - [Install Libraries and Set up Serving](#install-libraries-and-set-up-serving-3)
+    - [Supported models](#supported-models-1)
+    - [How to use in AgentScope](#how-to-use-in-agentscope-4)
+- [Model Inference API](#model-inference-api)
 
 ## Local Model API Serving
 
+### ollama
+
+[ollama] is a CPU inference engine for LLMs. With ollama, developers can build their local model API serving without GPU requirements.
```
Comment (qbc2016) [inline scripts/README.md:36]: Missing hyperlinks for [ollama]？

File: README.md [hunk 9]
```diff
@@ -12,66 +14,116 @@ English | [**中文**](README_ZH.md)
...
 | <img src="https://gw.alicdn.com/imgextra/i1/O1CN01hhD1mu1Dd3BWVUvxN_!!6000000000238-2-tps-400-400.png" width="100" height="100"> | <img src="https://img.alicdn.com/imgextra/i2/O1CN01tuJ5971OmAqNg9cOw_!!6000000001747-0-tps-444-460.jpg" width="100" height="100"> | <img src="https://img.alicdn.com/imgextra/i3/O1CN01UyfWfx1CYBM3WqlBy_!!6000000000092-2-tps-400-400.png" width="100" height="100"> |
 
 ----
 
 ## News
+- ![new](https://img.alicdn.com/imgextra/i4/O1CN01kUiDtl1HVxN6G56vN_!!6000000000764-2-tps-43-19.png)
+[2024-03-15] We release **AgentScope** v0.0.2 now! In this new version,
+AgentScope supports [DashScope](https://modelscope.github.io/agentscope/en/tutorial/203-model.html#supported-models), [ollama](https://modelscope.github.io/agentscope/en/tutorial/203-model.html#supported-models) and [Gemini](https://modelscope.github.io/agentscope/en/tutorial/203-model.html#supported-models) APIs.
```
[the remark this answers is not in the document]
Comment (pan-x-c) [reply README.md:31]: Currently, the link can only be directed to their common parent title.

**Review by DavdGao — APPROVED** [review]
LGTM

**Pull request merged**